Description: Sort recordings dirs first and by name, recs by date or name
 (if last char in dir is one of '.-$ª').
Author: FrankJepsen at vdrportal.de
Origin: http://www.jepsennet.de/vdr/download/11_sortrecordings.dpatch

--- a/recording.c
+++ b/recording.c
@@ -815,8 +815,8 @@
               s1 = t;
            }
         t++;
-        }
-  if (s1 && s2)
+        } *s1 = 255;
+  if (s1 && s2 && s1 != s && !strchr(".-$ª", *(s1 - 1)))
      memmove(s1 + 1, s2, t - s2 + 1);
   return s;
 }
@@ -824,7 +824,7 @@
 char *cRecording::SortName(void) const
 {
   if (!sortBuffer) {
-     char *s = StripEpisodeName(strdup(FileName() + strlen(VideoDirectory) + 1));
+     char *s = StripEpisodeName(strdup(FileName() + strlen(VideoDirectory) ));
      strreplace(s, '/', 'a'); // some locales ignore '/' when sorting
      int l = strxfrm(NULL, s, 0) + 1;
      sortBuffer = MALLOC(char, l);
